src/clients/near.ts

```diff
@@ -559,6 +605,372 @@ export class NearBridgeClient {
     }
   }
 
+  // =====================================================================
+  // BITCOIN BRIDGE METHODS (mirrors Rust SDK NearBridgeClient + BtcConnector)
+  // =====================================================================
+
+  /**
+   * Get Bitcoin deposit address from Satoshi Bridge (BTC -> NEAR flow start)
+   * Mirrors get_btc_address() from Rust SDK
+   */
+  async getBitcoinDepositAddress(
+    recipientId: string,
+    amount?: bigint,
+    fee?: bigint,
+  ): Promise<{ depositAddress: string; btcDepositArgs: BtcDepositArgs }> {
+    // Validate minimum amount if provided
+    if (amount) {
+      const bitcoinConfig = await this.getBitcoinBridgeConfig()
+      if (amount < BigInt(bitcoinConfig.min_deposit_amount)) {
+        throw new Error(
+          `Amount ${amount} is below minimum deposit amount ${bitcoinConfig.min_deposit_amount}`,
+        )
+      }
+    }
+
+    // Deposit msg depends on if the receiver is an Omni Address or not
+    let depositMsg: DepositMsg
+    if (recipientId.includes(":")) {
```

Comment:
This function is somewhat limited since it doesn’t allow creating transfers to deposit into other dapps eg Near-intents.